cmake/FindReadline.cmake

```diff
@@ -3,18 +3,23 @@
 #  READLINE_INCLUDE_DIR : e.g., /usr/include/.
 #  READLINE_LIBRARY     : Library path of readline library
 #  READLINE_FOUND       : True if found.
-FIND_PATH(READLINE_INCLUDE_DIR NAME readline/readline.h
+
+add_library(readline INTERFACE)
+
+find_path(READLINE_INCLUDE_DIR NAME readline/readline.h
     HINTS /usr/local/opt/readline/include/ /opt/homebrew/opt/readline/include/ $ENV{HOME}/local/include /opt/local/include /usr/local/include /usr/include)
 
-FIND_LIBRARY(READLINE_LIBRARY NAME readline
+find_library(READLINE_LIBRARY NAME readline
     HINTS /usr/local/opt/readline/lib/ /opt/homebrew/opt/readline/lib/ $ENV{HOME}/local/lib64 $ENV{HOME}/local/lib /usr/local/lib64 /usr/local/lib /opt/local/lib64 /opt/local/lib /usr/lib64 /usr/lib
```

Comment:
Not yours, but the first one is oddly specify compared to the other paths. Any ideas why?

Reply:
On macOS x64, the library and include paths are `/usr/local/opt/readline/lib` and `/usr/local/opt/readline/include`, respectively. If we don't specify this specific library/include directory path, CMake will use the library shown in Snippet 1, which is incompatible since it does not provide the `append_history` function.

Snippet 1
```
Found readline library: inc=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk/usr/include, lib=/usr/lib/libreadline.dylib
```

Snippet 2
```
In file included from /Users/hyrise-ci/.jenkins/workspace/hyrise_hyrise_PR-2422/clang-debug/src/bin/CMakeFiles/hyriseConsole.dir/Unity/unity_0.cxx:3:

/Users/hyrise-ci/.jenkins/workspace/hyrise_hyrise_PR-2422/src/bin/console/console.cpp:189:11: error: use of undeclared identifier 'append_history'

      if (append_history(1, _history_file.c_str()) != 0) {

          ^
```